### PR TITLE
Added a function to allow a user to set a gamma correction factor.

### PR DESCRIPTION
**Pull request by Romonaga**
This will generate and apply this correction table to the led gamma setting.

### CODE REVIEW

**Comment by Gadgetoid** [comment]
Sorry I haven't yet had a chance to give this a test, but thank you, it's in my backlog!

**Comment by Romonaga** [comment]
> Sorry I haven't yet had a chance to give this a test, but thank you, it's in my backlog!

that you even consider it is a honor.  As you see, it makes not attempt to alter the memory that is created, it simply replaces the values.  It has been working well in my code base and allows me to change gamma settings on the fly at will now.
Again, thanks for considering it, been using this lib for years, and would be fantastic to have even this small contribution.

File: SConstruct [hunk 1]
```diff
@@ -48,6 +48,7 @@ platforms = [
             ],
             'LINKFLAGS' : [
                 "-lrt",
+		"-lm",
```
Comment (Gadgetoid) [inline SConstruct:51]: Should be space-indented, rather than tabs (Probably because SCons is a Python tool)

File: ws2811.c [hunk 2]
```diff
@@ -1282,3 +1282,23 @@ const char * ws2811_get_return_t_str(const ws2811_return_t state)
 
     return "";
 }
+
+
+void setCustomGammaFactor(ws2811_t *ws2811, double gammaFactor)
```
Comment (Gadgetoid) [inline ws2811.c:1287]: Should probably be `ws2811_set_custom_gamma_factor`. Not exactly c-like, but will match the other function names for consistency.

I think `gammaFactor` should be `gamma_factor` too.

File: ws2811.h [hunk 3]
```diff
@@ -122,6 +122,7 @@ void ws2811_fini(ws2811_t *ws2811);                                    //< Tear
 ws2811_return_t ws2811_render(ws2811_t *ws2811);                       //< Send LEDs off to hardware
 ws2811_return_t ws2811_wait(ws2811_t *ws2811);                         //< Wait for DMA completion
 const char * ws2811_get_return_t_str(const ws2811_return_t state);     //< Get string representation of the given return state
+void setCustomGammaFactor(ws2811_t *ws2811, double gammaFactor); //Set a custom Gamma correction array based on a gamma correction factor
```
Comment (Gadgetoid) [inline ws2811.h:125]: Again, should probably be `ws2811_set_custom_gamma_factor`.

Indentation level and style of the inline comment should also match the other lines.

Comment (Romonaga) [reply ws2811.h]: New requests, with changes per code review.

**Comment by Gadgetoid** [comment]
Merged with the knowledge that this breaks the CMakeLists.txt build which I will fix shortly.

Thank you!

**Pull request merged**